gpgme.go

```diff
@@ -389,7 +390,11 @@ func (c *Context) GetKey(fingerprint string, secret bool) (*Key, error) {
 	key := newKey()
 	cfpr := C.CString(fingerprint)
 	defer C.free(unsafe.Pointer(cfpr))
-	return key, handleError(C.gpgme_get_key(c.ctx, cfpr, &key.k, cbool(secret)))
+	err := handleError(C.gpgme_get_key(c.ctx, cfpr, &key.k, cbool(secret)))
+	if key.k == nil {
```

Comment:
Not to my testing, this method simply return `nil, nil` when the key is not found. So there's no error nor key at the end. And we can't check for key nilness either in containers/image because the key structure  is correctly initialized but the inner C struct `k` isn't. 
This is just one of the many ways to fix this, if @mtrmac has other in mind, I'll change it. 
Note that this patch is however fixing the issue.